static/app/views/monitors/components/monitorHeaderActions.tsx

```diff
@@ -22,11 +23,19 @@ type Props = {
 
 function MonitorHeaderActions({monitor, orgId, onUpdate}: Props) {
   const api = useApi();
+  const router = useRouter();
+
   const {selection} = usePageFilters();
 
   const handleDelete = async () => {
     await deleteMonitor(api, orgId, monitor.slug);
-    browserHistory.push(normalizeUrl(`/organizations/${orgId}/crons/`));
+    browserHistory.push(
+      normalizeUrl(
+        `/organizations/${orgId}/crons/${
+          router.location.query.prevUrl ? `?project=${router.location.query.prevUrl}` : ''
```

Comment:
This method works, but adds unnecessary query parameters to the URL

It also only functions when coming from the monitor overview page, if you were to add a monitor and then delete it immediately you would be redirected back to a page without a project selection

What I think we could do here is leverage the `selection` from `usePageFilters` which holds global state around which project(s) were selected and feed those into the query params. Let me know if that makes sense and if it works out, I haven't tried it.

Reply:
Yes @davidenwang correct. Using prevUrl is not needed we need selected filters - projects and environments which will be present in usePageFilters.

And, yes i only covered just one flow but the creation, edition redirection should also be handled and also the breadcrumbs.

I am pushing a commit in sometime for this.

Thanks